### DISCUSSION TITLE
Directus filter/operator for Mysql SOUNDS LIKE Operator

### DISCUSSION THREAD

**Discussion by abuzahran** [directus/directus#2738] [Feature Requests]
Hi 

In the Directus API is there anyway I can use a filter similar to MySQL 'SOUNDS LIKE' operator ? 

This is my MySQL Query:

<img width="301" alt="Screenshot 2020-06-02 at 17 50 46" src="https://user-images.githubusercontent.com/47009227/83547920-d5903b00-a4fa-11ea-9fef-6fdefc8d6c66.png">

and I am hoping if there is something like this I can use:

GET > https://URL_TO_API/public/PROJECT/items/COLLECTIO_NAME?filter[FIELD_NAME][SOUNDS LIKE]='SEARCH_KEYWORD'

I get the below error:

<img width="477" alt="Screenshot 2020-06-02 at 18 02 52" src="https://user-images.githubusercontent.com/47009227/83548206-4cc5cf00-a4fb-11ea-9b01-2a604e1bc2a5.png">

**Comment by benhaynes** [comment]
A pretty cool feature idea, but probably not too likely to get implemented by the core team any time soon (since it's very much an edge case). Also, I don't know if `SOUNDEX` is included in all DB vendors we support, and I don't think we're ready to start adding conditional features per vendor. 

Definitely a good Custom API Endpoint example though!

**Reply by abuzahran** [reply]
Thanks @benhaynes I thought custom endpoint might be the way forward. I don't think anything about zend framework but willing to learn ... 

@ everyone - I checked the docs  for custom endpoint and the example seems a bit daunting at first sight .. is there any easy to follow example you can reference (on github etc) where I can see how a custom endpoint has been implemented using a custom mysql query?

Thanks in advance ..

**Reply by benhaynes** [reply]
No problem! Also, the upcoming v9 (out in 4-6 weeks) is in Laravel, so I wouldn't spend too much time on Zend. We're also looking to have a Node API version in the coming months (TBD).

I thought there was a custom endpoint example in the codebase, but if not we might want to add one.

**Comment by abuzahran** [comment]
@benhaynes Laravel sounds fantastic.

Having said that, I managed to build a custom end point and it is working but I have a small issue where the fields=*.*.* does not work ... please see below. Anyone able to assist please ? thanks
<br/>
Custom end point /search
<br/>
`<?php
use Directus\Application\Http\Request;
use Directus\Application\Http\Response;
use Directus\Services\ItemsService;
use Zend\Db\Adapter\Adapter;

return [
    '' => [
        'method' => 'GET',
        'handler' => function (Request $request, Response $response) {
            
            $container = \Directus\Application\Application::getInstance()->getContainer();
            $dbConnection = $container->get('database');
            $itemsService = new ItemsService($this);
            $params = $request->getQueryParams();
            $acl = $container->get('acl');
            $dramas = $itemsService->findAll('dramas', $params);
            
            if ($acl->canRead('dramas')) {
            
            $sql = "SELECT * FROM `dramas`";
            
            if (isset($params["filter"]["drama"])) {
                    $sql = $sql . " WHERE drama SOUNDS LIKE '" . $params["filter"]["drama"]."'";
                }
                
              // print_r($sql);
                $result = $dbConnection->query($sql, Adapter::QUERY_MODE_EXECUTE);
                $result = $result->toArray();
                
            }
            print_r($sql);
                
            $final = array_replace_recursive($dramas["data"], $result);

            return $response->withJson([
                'data' => $final,
            ]);
        }
    ],
];
`
<br/><br/>
When I do GET > http://URL_TO_API......./custom/search?fields=*.*.*
<br/><br/>
I get all records from 'dramas' collection and second/third tier fields from the relational fields (O2M)
<br/><br/>
However, when I add the filter, I get the filtered records (as expected) but NOT the second/third tier relational fields (fields=*.*.* does not work)
<br/><br/>
GET > http://URL_TO_API......./custom/search?fields=*.*.* & filter[drama]=keyword
<br/><br/>
Please can you assist? 
<br/><br/>
Thanks alot!

**Reply by benhaynes** [reply]
This is out of my depth, but someone on @directus/team might be able to help. If you don't get a response fast enough here, you can also try reaching out to the community on Slack: https://directus.chat

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
